stdlib/assembly.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: Apache-2.0
+
+// These are the helpers from:
+// https://github.com/ethereum/solidity/issues/1319
+// https://github.com/ethereum/solidity/issues/7317
+// https://github.com/ethereum/solidity/issues/9824#issuecomment-717983295
+
+assembly {
+  // allocates a memory block
+  // TODO: use memoryguard
+  function alloc(size) -> ptr {
```

Comment:
```suggestion
  function allocate(size) -> ptr {
```

Reply:
I think actually these memory functions may be better as being part of the inline assembly dialect (as discussed on #10399).

Reply:
Yes, but I think it would be good to also have the translation between the two dialects available.